Building with closures and this keyword/main.js

```diff
@@ -0,0 +1,65 @@
+const clickMeBtn = document.getElementById('click-me-btn')
+const Person = {
+    name: 'John Doe',
+    age: 30,
+    greet() {
+        console.log(`Hello, my name is ${this.name} and my age is ${this.age}`);
+    }
+}
+
+Person.greet(); // Output: Hello, my name is John Doe and my age is 30
+
+Person.greet.call({name: 'Jane Doe', age: 25}); // Output: Hello, my name is Jane Doe and my age
+
+Person.greet.apply({name: 'Jane Doe', age: 25}); // Output: Hello, my name is Jane Doe and my age
+
+Person.greet.bind({name: 'John Doe', age: 25})(); // Output: Hello, my name is Jane Doe and my age
+
+
+function handleClick() {
+    console.log(this.id)
+    console.log(this.textContent)
+}
+
+
+// const handleClick = (e) => {
+//     console.log(this.id)
+//     console.log(this.textContent)
+// } // TypeError: Cannot read properties of undefined (reading 'id')
+
+
+function createCounter() {
+    let count = 0;
```

Comment:
- Usage of `this.count` is incorrect
- If the idea is to simulate oop methods using the function then each method should have a single responsibility e.g. separate method for logging.

Reply:
The usage was part of the instructions. We were told to use it that way to observe the behavior of the this keyword